modules/general_functions.py

```diff
@@ -656,3 +657,51 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+# FIXME: Find better placement
+class ReferenceDensity:
+    def __init__(self, layers: List = None):
+        self.layers = layers
+        self.reference_density = 0.0
+        self.layer_reference_density = 0.0
+        self.reference_density_limit = 10.0
+        self.reference_density_meter = 10.0
```

Comment:
`reference_density_meter` isn't a good name because it's not reference density in meters.

Reply:
OK